crates/driver/src/infra/observe/mod.rs

```diff
@@ -61,7 +61,7 @@ pub fn fetching_liquidity_failed(err: &boundary::Error) {
 
 /// Observe the solutions returned by the solver.
 pub fn solutions(solutions: &[Solution]) {
-    if !solutions.is_empty() {
+    if !solutions.iter().any(|s| !s.is_empty()) {
```

Comment:
We want to print the log if any of the solutions is not empty, right?
```suggestion
    if solutions.iter().any(|s| !s.is_empty()) {
```